File: example-java-spring/src/main/java/de/qualersoft/robotframework/library/example/keywords/functions/FunctionKwds.java
package de.qualersoft.robotframework.library.example.keywords.functions;

import de.qualersoft.robotframework.library.annotation.Keyword;
import de.qualersoft.robotframework.library.annotation.KwdArg;
import de.qualersoft.robotframework.library.model.ParameterKind;
import jakarta.inject.Named;

import java.util.List;
import java.util.Map;

@Named
public class FunctionKwds {

  @Keyword
  public void fnCallWithVarargs(
      String posArg,
      @KwdArg(kind = ParameterKind.VARARG)
      List<String> varargs
  ) {
    System.out.printf("Mandatory posArg: %s%n", posArg);

    if (varargs.isEmpty()) {
      System.out.println("No varargs passed.");
    } else {
      System.out.printf("Got varargs: %s%n", varargs);
    }
  }

  @Keyword
  public void fnCallWithKwargs(
      String posArg,
      @KwdArg(kind = ParameterKind.KWARG)
      Map<String, String> kwargs
  ) {
    System.out.printf("Mandatory posArg: %s%n", posArg);

    if (kwargs.isEmpty()) {
      System.out.println("No kwargs passed.");
    } else {
      System.out.printf("Got kwargs: %s%n", kwargs);
    }
  }

  @Keyword
  public void fnCallWithVarAndKwArgs(
      String posArg,
      @KwdArg(kind = ParameterKind.VARARG)
      List<String> varargs,
      @KwdArg(kind = ParameterKind.KWARG)
      Map<String, String> kwargs
  ) {
    System.out.printf("Mandatory posArg: %s%n", posArg);

    if (varargs.isEmpty()) {
      System.out.println("No varargs passed.");
    } else {
      System.out.printf("Got varargs: %s%n", varargs);
    }

    if (kwargs.isEmpty()) {
      System.out.println("No kwargs passed.");
    } else {
      System.out.printf("Got kwargs: %s%n", kwargs);
    }
  }
}

File: example-kotlin-spring/src/main/kotlin/de/qualersoft/robotframework/library/example/keywords/functions/FunctionKwds.kt
package de.qualersoft.robotframework.library.example.keywords.functions

import de.qualersoft.robotframework.library.annotation.Keyword
import de.qualersoft.robotframework.library.annotation.KwdArg
import de.qualersoft.robotframework.library.model.ParameterKind
import jakarta.inject.Named

@Named
open class FunctionKwds {

  @Keyword
  fun fnCallWithVarargs(
    posArg: String,
    @KwdArg(kind = ParameterKind.VARARG)
    varargs: List<String>
  ) {
    println("Mandatory posArg: $posArg")

    if (varargs.isEmpty()) {
      println("No varargs passed.")
    } else {
      println("Got varargs: $varargs")
    }
  }

  @Keyword
  fun fnCallWithKwargs(
    posArg: String,
    @KwdArg(kind = ParameterKind.KWARG)
    kwargs: Map<String, String>
  ) {
    println("Mandatory posArg: $posArg")

    if (kwargs.isEmpty()) {
      println("No kwargs passed.")
    } else {
      println("Got kwargs: $kwargs")
    }
  }

  @Keyword
  fun fnCallWithVarAndKwArgs(
    posArg: String,
    @KwdArg(kind = ParameterKind.VARARG)
    varargs: List<String>,
    @KwdArg(kind = ParameterKind.KWARG)
    kwargs: Map<String, String>
  ) {
    println("Mandatory posArg: $posArg")

    if (varargs.isEmpty()) {
      println("No varargs passed.")
    } else {
      println("Got varargs: $varargs")
    }

    if (kwargs.isEmpty()) {
      println("No kwargs passed.")
    } else {
      println("Got kwargs: $kwargs")
    }
  }
}
